File: atvcarro/carro.py

class carro:
	def __init__(self):
		self.maxGasolina = 10
		self.tanque = 0
		carro.km = 0
		self.lugaresOcupados = 0
		self.maxLugaresOcupados = 2

	def __str__(self):
		return ("\npass: " + str(self.lugaresOcupados)  +
		", gas: " + str(self.tanque) +
		", km: " + str(self.km))
		
	def embarcar(self):
		if self.lugaresOcupados < self.maxLugaresOcupados:
			self.lugaresOcupados = self.lugaresOcupados+1
			return True 
		else:
			print ("fail: limite de pessoas atingido")
			
	def desembarcar(self):
		if self.lugaresOcupados > 0:
			self.lugaresOcupados = self.lugaresOcupados-1
			return True
		else:
			print ("fail: nao ha ninguem no carro")
		
	def abastecer(self, qtdGasolina):
		if self.tanque < self.maxGasolina:
			self.tanque = self.tanque+qtdGasolina
		if self.tanque > self.maxGasolina:
			self.tanque = self.maxGasolina 
			
	def dirigir(self, qtdKm):
		totalQuilometragem = self.tanque * 10 #quantos quilomntros o carro pode fazer com a gasolina atual do tanque
		gastoGasolina = qtdKm/10
		if((self.lugaresOcupados > 0) and (qtdKm <= totalQuilometragem)):
			self.km = self.km + qtdKm
			self.tanque = self.tanque - gastoGasolina
			return True
		elif qtdKm > totalQuilometragem:
			print ("fail: nao ha combustivel suficiente para realizar a viagem")
		if(self.lugaresOcupados  == 0): 
			print ("fail: nao ha ninguem para realizar a viagem")

			

		
		
		
		  
	
fusca = carro()# criando o objeto fusca da classe carro
#fusca.km = 8  # alterando o valor do atributo do objeto 

line = ""
print ("embarcar, desembarcar, abastecer, dirigir, show, end")
while (line!="end"):
	line = input()
	ui = line.split(" ")
	if ui[0] == "embarcar":
		fusca.embarcar()
	elif ui[0] == "desembarcar":
		fusca.desembarcar()
	elif ui[0] == "abastecer":
		fusca.abastecer(int(ui[1]))
	elif ui[0] == "dirigir":
		fusca.dirigir(int(ui[1]))
	elif ui[0] == "show":
		print(fusca)
	elif ui[0] == "end":
		break
	else:
		print ("comando invalido")

File: Carro/src/Carro.java

	import java.util.Scanner;

	public class Carro {
		float tanque; // atributo p uma numeracao do tipo inteiro //
		float maxtanque;
		int qtdpass; 
		int maxqtdpass;
		float km; 
		
		public Carro() { // construtor do carro 
			tanque = (float) 0.0;
			maxtanque = (float) 10.0;
			maxqtdpass = 2;
			qtdpass = 0;
			km = (float) 0.0;
		}
		
		public void show() {
			System.out.println("pass:" +qtdpass+ ", gas:" +tanque+  ", km:"+km);
		}
		public void embarcar(){
			if (qtdpass < maxqtdpass) {
				qtdpass += 1;
			}else{
				System.out.println("fail: limite de pessoas atingido");
			}
			
		}
		public void desembarcar() {
			if (qtdpass > 0) {
				qtdpass -= 1;
			}else{
				System.out.println("fail: nao ha ninguem dentro do carro");
			}
		}
		public void abastecer(float qtdgasolina) {
			if (tanque < maxtanque) {
				tanque += qtdgasolina;
				if (tanque > maxtanque) {
					tanque = maxtanque;
				}
			}
		}
		
		public void dirigir(float qtdkm) {
			float gastogasolina;
			gastogasolina = qtdkm / 10;
			if((qtdpass > 0) && (gastogasolina <= tanque)){
				tanque -= gastogasolina;
				km += qtdkm;
			}else if(gastogasolina > tanque) {
				System.out.println("fail: gasolina insuficiente");
				
			}else if(qtdpass <= 0) {
				System.out.println("fail: o carro nao tem ninguem");
			}
			
		}
		
		public static void main(String[] args) {
			Carro fusca = new Carro(); // criando objeto ou instanciando um objeto//
			Scanner input = new Scanner(System.in);
			boolean entrar = true;
			System.out.println("embarcar, desembarcar, abastecer valor, dirigir valor, sair");
			while(entrar) {
				String linha = input.nextLine();
				String[] comando = linha.split(" ");
				String op = comando[0];
				if(op.equals("embarcar")) {
					fusca.embarcar();
				}else if(op.equals("desembarcar")) {
					fusca.desembarcar();		
				}else if(op.equals("show")){
					fusca.show();				
				}else if(op.equals("abastecer")) {
					float qtdgasolina = Float.parseFloat(comando[1]);
					fusca.abastecer(qtdgasolina);
				}else if(op.equals("dirigir")) {
					float qtdkm = Float.parseFloat(comando[1]);
					fusca.dirigir(qtdkm);
				}else if(op.equals("sair")) {	
					entrar = false;
				}
			}
		}
	}


